src/components/Extension/CloudinaryExtension/utils/CloudinaryService.tsx

```diff
@@ -0,0 +1,28 @@
+import {
+  Configuration,
+  Cloudinary as CoreCloudinary,
+  Util,
+} from 'cloudinary-core'
+import { type UploadWidgetOptions } from './types'
+
+export const url = (publicId: string, options: Configuration.Options) => {
+  try {
+    const scOptions = Util.withSnakeCaseKeys(options)
+    const cl = CoreCloudinary.new(options)
+    return cl.url(publicId, scOptions)
+  } catch (e) {
+    return null
+  }
+}
+
+export const openUploadWidget = (
+  options: UploadWidgetOptions,
+  callback: (error: unknown, result: Record<string, unknown>) => void
+): { open: () => void } => {
+  if (window?.cloudinary === undefined) {
+    throw new Error('Cloudinary is not instantiated on window.')
+  }
+
+  //@ts-expect-error
```

Comment:
[C] Same here, if it can be removed - let's remove it. If not, let's add an explanation.